AppKit/CPSplitView.j

```diff
@@ -1453,45 +1787,63 @@ var CPSplitViewDelegateKey          = "CPSplitViewDelegateKey",
     subviews, otherwise the parent restore will also change the positioning of the child.
 
     */
-    if (_autosaveName)
-    {
-        // Schedule /before/ [super initWithCoder:]. This way this instance's _restoreFromAutosaveIfNeeded
-        // will happen before that of any subviews loaded by [super initWithCoder:].
-        [[CPRunLoop currentRunLoop] performSelector:@selector(_restoreFromAutosaveIfNeeded) target:self argument:nil order:0 modes:[CPDefaultRunLoopMode]];
-    }
+
+    // As subviews are often not fully ready to use at this time, we'll try to finalize the initialization after all views initializations are done
+    [[CPRunLoop currentRunLoop] performSelector:@selector(_finalizeInitWithCoder) target:self argument:nil order:0 modes:[CPDefaultRunLoopMode]];
 
     self = [super initWithCoder:aCoder];
 
     if (self)
     {
+        [self _init];
+
         _suppressResizeNotificationsMask = 0;
         _preCollapsePositions = [CPMutableDictionary new];
 
         _currentDivider = CPNotFound;
-        _shouldAutosave = YES;
+        _shouldAutosave = NO;
         _isTracking = NO;
 
-        _DOMDividerElements = [];
-
-        _buttonBars = [aCoder decodeObjectForKey:CPSplitViewButtonBarsKey] || [];
+        _buttonBars = [aCoder decodeObjectForKey:CPSplitViewButtonBarsKey] || @[];
 
         [self setDelegate:[aCoder decodeObjectForKey:CPSplitViewDelegateKey]];
 
-        _isPaneSplitter = [aCoder decodeBoolForKey:CPSplitViewIsPaneSplitterKey];
         [self _setVertical:[aCoder decodeBoolForKey:CPSplitViewIsVerticalKey]];
 
-        if (_autosaveName)
-        {
-            [self _restoreFromAutosave];
-            // Remember the frame size we had at this point so that we can restore again if it changes
-            // before the next runloop cycle. See above notes.
-            _shouldRestoreFromAutosaveUnlessFrameSize = [self frameSize];
-        }
+        _dividerSubviews  = [aCoder decodeObjectForKey:CPSplitViewDividerSubviewsKey]  || @[];
+        _arrangedSubviews = [aCoder decodeObjectForKey:CPSplitViewArrangedSubviewsKey] || @[];
+        _realSubviews     = [aCoder decodeObjectForKey:CPSplitViewRealSubviewsKey]     || @[];
+
+        [self setArrangesAllSubviews:[aCoder decodeBoolForKey:CPSplitViewArrangesAllSubviewsKey]];
+        [self setDividerStyle:[aCoder decodeIntForKey:CPSplitViewDividerStyleKey]];
+
+        _isCSSBased = [[self theme] isCSSBased];
+
+        // Final operations will be performed in _finalizeInitWithCoder
     }
 
     return self;
 }
 
+- (void)_finalizeInitWithCoder
```

Comment:
Can't we just rename this method to `- (id)_cibInstantiate` and add a `return self;` at the end of it? With that we could remove the `performSelector:` on the runloop above? It will be called upon after all the views and objects in the cib are initialized.

Reply:
New commit implementing your suggestion. Works in my testing.